pages/cookbook/jettons.mdx

```diff
@@ -1,9 +1,156 @@
+Note, that notifications are coming from YOUR contract's jetton wallet, so [`myAddress()`](/ref/core-common#myaddress) should be used in owner address field.
+Wallet initial data layout is shown below, but sometimes it can differ.
+
+```tact
+struct JettonWalletData {
+    balance: Int as coins;
+    ownerAddress: Address;
+    jettonMasterAddress: Address;
+    jettonWalletCode: Cell;
+}
+
+fun calculateJettonWalletAddress(ownerAddress: Address, jettonMasterAddress: Address, jettonWalletCode: Cell): Address {
+    let initData = JettonWalletData{
+        balance: 0,
+        ownerAddress,
+        jettonMasterAddress,
+        jettonWalletCode,
+    };
+
+    return contractAddress(StateInit{code: jettonWalletCode, data: initData.toCell()});
+}
+
+contract Sample {
+    jettonWalletCode: Cell;
+    jettonMasterAddress: Address;
+
+    init(jettonWalletCode: Cell, jettonMasterAddress: Address) {
+        self.jettonWalletCode = jettonWalletCode;
+        self.jettonMasterAddress = jettonMasterAddress;
+    }
+
+    receive(msg: TransferNotification) {
+        let myJettonWalletAddress = calculateJettonWalletAddress(myAddress(), self.jettonMasterAddress, self.jettonWalletCode);
+        require(sender() == myJettonWalletAddress, "Notification not from your jetton wallet!");
+
+        // your logic of processing token notification
+    }
+}
+```
+
+## Sending jetton transfer
+
+To send jetton transfer use `send` or similar [function](/book/send).
+Note that `myJettonWalletAddress` may also be stored in contract storage to use less gas in every transaction.
+
+```tact
+message(0xf8a7ea5) TokenTransfer {
+    queryId: Int as uint64;
+    amount: Int as coins;
+    destination: Address;
+    excessesDestination: Address;
+    customPayload: Cell? = null;
+    forwardTonAmount: Int as coins;
+    forwardPayload: Slice as remaining;
+}
+
+receive("send") {
+    let myJettonWalletAddress = calculateJettonWalletAddress(myAddress(), self.jettonMasterAddress, self.jettonWalletCode);
+    send(SendParameters{
+        to: myJettonWalletAddress,
+        value: ton("0.05"),
+        body: TokenTransfer{
+            queryId: 42,
+            amount: jettonAmount, // jetton amount you want to transfer
+            destination: msg.userAddress, // address you want to transfer jettons. Note that this is address of jetton wallet owner, not jetton wallet itself
+            excessesDestination: msg.userAddress, // address were left toncoins will go
+            customPayload: null, // in most cases will be null and can be omited. Needed for custom logic on jettons itself
+            forwardTonAmount: 1, // amount that will be transferred with TransferNotification. Needed for custom logic execution like in example below
+            forwardPayload: beginCell().storeBool(false).endCell().beginParse(), // TODO: this is so gas consuming, maybe there is another way? Or use generic (TBD)? Or there is better way to just left it
```

Comment:
You might try to pre-compute the Cell with that single bit as a Base64 string, then put it into the comptime function [`cell()`](https://docs.tact-lang.org/ref/core-comptime#cell), then use `.beginParse()` on it:

```tact
// beginCell().storeBool(false).endCell(),  but precomputed!
forwardPayload: cell("te6cckEBAQEAAwAAAUD20kA0").beginParse(),
````

In 1.5.0 there would be a similar comptime function `slice()`, which would be even more handy (no need for `.beginParse()`)

Reply:
I think it would be nicer to have smth like `builder as remaining` to have possibility to store builder not cell|slice

Reply:
But nvm, will leave that as it is